queue/abstract/queue_session.lua

```diff
@@ -0,0 +1,194 @@
+local log      = require('log')
+local fiber    = require('fiber')
+local uuid     = require('uuid')
+
+local util     = require('queue.util')
+local qc       = require('queue.compat')
+local num_type = qc.num_type
+local str_type = qc.str_type
+
+
+local queue_session = {
+    settings = {}
+}
+
+-- Sessions that don't have any active connections.
+local inactive_sessions = {}
+
+--- Create everything that's needed to work with "shared" sessions.
+local function identification_init()
+    local queue_session_ids = box.space._queue_session_ids
+    if queue_session_ids == nil then
+        queue_session_ids = box.schema.create_space('_queue_session_ids', {
+            temporary = true,
+            format = {
+                {name = 'box_session_id', type = num_type()},
+                {name = 'session_uuid', type = str_type()}
+            }
+        })
+
+        queue_session_ids:create_index('box_sid', {
+            type = 'tree',
+            parts = {1, num_type()},
+            unique = true
+        })
+        queue_session_ids:create_index('uuid', {
+            type = 'tree',
+            parts = {2, str_type()},
+            unique = false
+        })
+    end
+end
+
+--- Get session UUID by connection ID.
+local function get_uuid(connection_id)
+    local session_ids = box.space._queue_session_ids:get(connection_id)
+    local session_uuid
+    if session_ids ~= nil then
+        session_uuid = session_ids[2]
+    else
+        session_uuid = queue_session.identify(connection_id)
+    end
+
+    return session_uuid
+end
+
+local function cleanup_inactive_sessions()
+    local cur_time = util.time()
+    for session_uuid, exp_time in pairs(inactive_sessions) do
+        if cur_time >= exp_time then
+            if queue_session._on_session_remove ~= nil then
+                queue_session._on_session_remove(session_uuid)
+            end
+            inactive_sessions[session_uuid] = nil
+        end
+    end
+end
+
+--- Create an expiration fiber to cleanup expired sessions.
+local function create_expiration_fiber()
+    local exp_fiber = fiber.create(function()
```

Comment:
I think we should add names for all fibers that we create.

fiber.self():name('queue.session_expiration_fiber')

Reply:
Done.